server/src/main/java/org/elasticsearch/index/mapper/AbstractScriptFieldType.java

```diff
@@ -217,6 +218,12 @@ public void validateMatchedRoutingPath(final String routingPath) {
         );
     }
 
+    @Override
+    public boolean fieldHasValue(FieldInfos fieldInfos) {
+        // We consider script filed types to always have value.
```

Comment:
s/filed/field  and perhaps explain why: they have no footprint in lucene, and running the script is necessary to know if they have a value, but that is expensive hence field caps considers them always non - empty

Reply:
Thanks, corrected and updated comment.